### PR TITLE
Adding difftime to date truncates the difftime to full dates

### PR DESCRIPTION
**Pull request by krlmlr**
For full _hms_ support. Do we need a lossy cast warning?

### CODE REVIEW

**Comment by hadley** [comment]
Yes, we need a warning here.

File: R/type-date-time.R [hunk 1]
```diff
@@ -427,24 +427,34 @@ vec_arith.difftime.POSIXct <- function(op, x, y) {
 #' @method vec_arith.Date difftime
 #' @export
 vec_arith.Date.difftime <- function(op, x, y) {
+  # Exit early to avoid warning
+  if (!(op %in% c("+", "-"))) {
```
Comment (hadley) [inline R/type-date-time.R]: I don't like moving this up front. Instead could we make a wrapper function, something like this:

```R
round_warn <- function(x, to) {
  x_floor <- floor(x)
  report_lossy_cast(x, to, x != x_floor)
  x_floor
}
```

Then could call instead `vec_arith_base()`?

Comment (krlmlr) [reply R/type-date-time.R]: Do you mind the fallthrough I've added?

Comment (hadley) [reply R/type-date-time.R]: Yeah, that's fine.

**Pull request merged**